packages/astro/client.d.ts

```diff
@@ -111,9 +112,7 @@ declare module '*.avif' {
 	export default metadata;
 }
 declare module '*.svg' {
-	type Props = astroHTML.JSX.SVGAttributes;
-
-	const Component: ((_props: Props) => any) & ImageMetadata;
+	const Component: SvgComponent;
```

Comment:
Here the type is scoped so it doesn't leak. Also I think we should move the ImageMetadata usage here:

```suggestion
  type SvgComponent = import('./dist/assets/types.js').SvgComponent;
	const Component: SvgComponent & ImageMetadata;
```

Reply:
What's the reason for combining `ImageMetadata` and not below in `export type`?

Reply:
That's because `ImageMetadata` is not part of the `SvgComponent`, it just so happens that `.svg` in Astro are both components **and** image metadata. I hope that makes sense!